.travis.yml

```diff
@@ -24,7 +20,22 @@ script:
 after_success:
   - codecov
 
-matrix:
+jobs:
   include:
+    # Default stage: test
     - python: 3.7
       env: TEST_LINT=1
+    - python: nightly
+    - python: 3.7
+    - python: 3.6
+    - python: 3.5
+    # Only deploy if all test jobs passed
+    - stage: deploy
+      python: 3.7
+      if: tag IS present
+      deploy:
+        provider: pypi
+        user: __token__
+        password: $PYPI_TOKEN
+        on:
+          tags: true
```

Comment:
I don't think it is relevant to have the on: tags: true section assuming we have `if: tag IS present` already. TravisCI is very hard to comprehend though so I'm certainly not certain.